ms-common-auth/src/main/java/net/trajano/ms/authz/ClientInfoResource.java

```diff
@@ -0,0 +1,76 @@
+package net.trajano.ms.authz;
+
+import io.swagger.annotations.Api;
+import net.trajano.ms.auth.spi.ClientValidator;
+import net.trajano.ms.core.ErrorResponses;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import javax.annotation.security.PermitAll;
+import javax.ws.rs.GET;
+import javax.ws.rs.HeaderParam;
+import javax.ws.rs.Path;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.Response;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+
+/**
+ * Client info endpoint resource.
+ *
+ * @author Archimedes Trajano
+ */
+@Api
+@Path("/client")
+@PermitAll
+public class ClientInfoResource {
+
+    @Autowired
+    private ClientValidator clientValidator;
+
+    @GET
+    public Response validateClientInfo(@HeaderParam("Origin") final String origin,
+        @HeaderParam("Referer") final String referrer,
+        @HeaderParam(HttpHeaders.AUTHORIZATION) final String authorization) {
+
+        if (origin == null && referrer == null) {
+            throw ErrorResponses.invalidRequest("Missing Origin");
+        }
+        if (authorization == null) {
+            throw ErrorResponses.invalidRequest("Missing Authorization");
+        }
+        final URI originUri;
+        try {
+            final URL tempOrigin;
+            if (origin == null && referrer != null) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2589)